erpnext/accounts/report/general_ledger/general_ledger.py

```diff
@@ -87,7 +91,7 @@ def set_account_currency(filters):
 		account_currency = None
 
 		if filters.get("account"):
-			account_currency = get_account_currency(filters.account)
+			account_currency = get_account_currency(filters.account[0])
```

Comment:
Instead of simply taking the first account it should check if only one account is selected or all selected accounts have the same currency then apply this

Reply:
Okay, but what should we do if neither of those condition are met?

Reply:
Nothing, just let the filters be same, no change

Reply:
Done.